go-controller/pkg/ovn/egressip.go

```diff
@@ -1670,39 +1712,71 @@ func (e *egressIPZoneController) getTransitIP(nodeName string, wantsIPv6 bool) (
 	return nodeTransitIP.IP.String(), nil
 }
 
-// createReroutePolicyOps creates an operation that does idempotent updates of the
-// LogicalRouterPolicy corresponding to the egressIP status item, according to the
-// following update procedure:
-// - if the LogicalRouterPolicy does not exist: it adds it by creating the
-// reference to it from ovn_cluster_router and specifying the array of nexthops
-// to equal [gatewayRouterIP]
-// - if the LogicalRouterPolicy does exist: it adds the gatewayRouterIP to the
-// array of nexthops
-// This function should be called with lock on nodeZoneState cache key status.Node
-func (e *egressIPZoneController) createReroutePolicyOps(ops []ovsdb.Operation, podIPNets []*net.IPNet, status egressipv1.EgressIPStatusItem, egressIPName string) ([]ovsdb.Operation, error) {
-	isEgressIPv6 := utilnet.IsIPv6String(status.EgressIP)
+func (e *egressIPZoneController) getNextHop(egressNodeName, egressIP, egressIPName string, isLocalZoneEgressNode, isOVNManaged bool) (string, error) {
 	var nextHopIP string
 	var err error
+	isEgressIPv6 := utilnet.IsIPv6String(egressIP)
 	// NOTE: No need to check if status.node exists or not in the cache, we are calling this function only if it
 	// is present in the nodeZoneState cache. Since we call it with lock on cache, we are safe here.
-	isLocalZoneEgressNode, _ := e.nodeZoneState.Load(status.Node)
 	if isLocalZoneEgressNode {
-		gatewayRouterIP, err := e.getGatewayRouterJoinIP(status.Node, isEgressIPv6)
-		if err != nil {
-			return nil, fmt.Errorf("unable to retrieve gateway IP for node: %s, protocol is IPv6: %v, err: %w", status.Node, isEgressIPv6, err)
+		if isOVNManaged {
+			gatewayRouterIP, err := e.getGatewayRouterJoinIP(egressNodeName, isEgressIPv6)
+			if err != nil {
+				return "", fmt.Errorf("unable to retrieve gateway IP for node: %s, protocol is IPv6: %v, err: %w",
+					egressNodeName, isEgressIPv6, err)
+			}
+			nextHopIP = gatewayRouterIP.String()
+		} else {
+			mgntPort := &nbdb.LogicalSwitchPort{Name: fmt.Sprintf("k8s-%s", egressNodeName)}
+			mgntPort, err := libovsdbops.GetLogicalSwitchPort(e.nbClient, mgntPort)
+			if err != nil && !apierrors.IsNotFound(err) {
+				return "", fmt.Errorf("failed to get next hop IP for non-ovn managed network and egress IP %s for node %s "+
+					"because unable to get management port: %v", egressIPName, egressNodeName, err)
+			} else if err != nil {
+				klog.Warningf("While attempting to get next hop for Egress IP %s (%s), failed to get management switch port", egressIPName, egressIP)
+				return "", nil
```

Comment:
if a create function calls this function and gets back this...then shouldn't it treat that as an error and retry later? At least this should logged as an error right? If egress IP is assigned to a node and it does not have a mangement port it must be some race condition and it is about to be configured.

Reply:
I added the doc string. Please inspect this nextHop fn again. 
For add path of EIP it will retry - for the delete, it will not and just clear OVN obj.